### DISCUSSION TITLE
UDP Queue - number of packets queued?

### DISCUSSION THREAD

**Discussion by palmerr23** [ssilverman/QNEthernet#84] [Q&A] [answered]
Is there any way to tell how many UDP packets are queued when the constructor is  EthernetUDP udp{QNE_PKT_QUEUE_LEN}  ?

As well, knowing that there were dropped packets since the last udp.parsepacket() call would be nice.

**Comment by ssilverman** [comment]
I can add those, but I need to put some thought into what I want to call the methods.

**Reply by ssilverman** [reply]
I’m thinking of changing “size” to “capacity”, the total queue size, and using “size” for how many are in the queue. This coincides with how `std::vector` names things.

**Comment by palmerr23** [comment]
Shawn,

My two cent's worth...

Yes, it will be good to be clear whether the call relates to current (parsePacket), next (peek) or total queue. In the current doco, it takes some careful reading to understand the meaning of the various uses of 'size' In choosing terms. 

For instance, changing the word might help clarify if receiveQueueSize() returns the maximum, or current, length of the queue. 

Capacity is useful for referring  to the maximum possible length, rather than the current number of queued packets. Using 'queueSize' for the current queue length could help distinguish it from the size() method of a parsed packet.

**Comment by ssilverman** [comment, marked as answer]
I changed size->capacity and the new size means "how full". I also added `droppedReceiveCount()` and `totalReceiveCount()` functions to both `EthernetUDP` and `EthernetFrame`. Check out the latest pushes.

Note that these counts are at the layer above the driver. To access the driver-layer dropped packets, you need to use the `LINK_STATS` option.

**Reply by ssilverman** [reply]
@palmerr23 feedback is welcome.

**Comment by palmerr23** [comment]
Thanks, I will try out the UDP versions and provide any insights I find.
